src/FsAutoComplete.Core/Project.fs

```diff
@@ -23,23 +83,47 @@ type Project (projectFile, onChange: ProjectFilePath -> unit) =
                 r.Reply response
                 return! loop (lastWriteTime, response)
             | SetResponse r ->
-                return! loop (lastWriteTime, r)
-        }    
-        loop (File.GetLastWriteTimeUtc projectFile, None)
+                persistentCache.SaveCache(lastWriteTime, r)
 
-    let fullPath = Path.GetFullPath projectFile
+                return! loop (lastWriteTime, r)
+        }
+        let lwt =
+            if File.Exists projectAsset then
+                let a = File.GetLastWriteTimeUtc projectAsset
+                let p = File.GetLastWriteTimeUtc projectFile
+                if a > p then a else p
+            else
+                File.GetLastWriteTimeUtc projectFile
+        let state = persistentCache.LoadCache lwt
```

Comment:
@Krzysztof-Cieslak  the invalidation logic is based just on the timestamp? timestamp are annoying, git change these too.

Maybe is better do an hash (fast like md5) of all known files who are important (assets + projectfile)? and paket stuff like reference and the temp `obj\paket.references`

This will not be enough, because doesnt cover the import of other msbuild files (like directory.build.props/targets). i can maybe check what files are important to check, at the project parsing step. i'll try fix that later.

For example the fsproj can change and project.assets.json remain the same, but project info need to be invalidated. Scenario: i change a Property in fsproj (this doesnt change the `project.assts.json`) that will conditionall include some compiles or compiler arguments. hash check cover that, timestamp not